ui/pages/confirmations/confirmation/alerts/useAlertsActions.tsx

```diff
@@ -0,0 +1,47 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+
+import { AlertActionKey } from '../../../../components/app/confirm/info/row/constants';
+import { getMemoizedUnapprovedConfirmations } from '../../../../selectors';
+import { useConfirmationNavigation } from '../../hooks/useConfirmationNavigation';
+
+export const useAlertsActions = (
+  hideAlertModal: () => void,
+  pendingConfirmation: ApprovalRequest<{ id: string }>,
+) => {
+  const pendingConfirmations = useSelector(getMemoizedUnapprovedConfirmations);
+
+  const { getIndex, navigateToIndex } = useConfirmationNavigation();
+
+  const processAction = useCallback(
+    (actionKey: string) => {
+      switch (actionKey) {
+        case AlertActionKey.ShowPendingConfirmation: {
+          const { id, origin } = pendingConfirmation;
```

Comment:
Minor, could we have a separate callback for each switch case to break up the logic?

Reply:
fixed